### PR TITLE
[multistage] follow up leaf scheduler

### PR DESCRIPTION
**Pull request by walterddr**
this is a follow-up on comments on #10711

- with the execution model change, mailbox send no longer need to do a while loop
- with the changes to leaf-stage operator it is no longer needed for a separate scheduler.

### CODE REVIEW

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java [hunk 1]
```diff
@@ -125,24 +126,25 @@ public String toExplainString() {
 
   @Override
   protected TransferableBlock getNextBlock() {
+    boolean canContinue = true;
```
Comment (walterddr) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java:129]:
```suggestion
```

File: pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/MailboxSendOperatorTest.java [hunk 2]
```diff
@@ -150,6 +150,12 @@ public void shouldSendDataBlock()
     MailboxSendOperator mailboxSendOperator = getMailboxSendOperator();
     TransferableBlock block = mailboxSendOperator.nextBlock();
 
+    // Then:
+    assertSame(block, dataBlock, "expected EOS block to propagate");
```
Comment (walterddr) [inline pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/MailboxSendOperatorTest.java:154]:
```suggestion
    assertSame(block, dataBlock, "expected data block to first return");
```

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java [hunk 3]
```diff
@@ -48,20 +48,16 @@ public class OpChainSchedulerService extends AbstractExecutionThreadService {
    * Default cancel signal retention, this should be set to several times larger than
    * {@link org.apache.pinot.query.service.QueryConfig#DEFAULT_SCHEDULER_RELEASE_TIMEOUT_MS}.
    */
-  private static final long DEFAULT_SCHEDULER_CANCELLATION_SIGNAL_RETENTION_MS = 60_000L;
+  private static final long SCHEDULER_CANCELLATION_SIGNAL_RETENTION_MS = 60_000L;
 
   private final OpChainScheduler _scheduler;
   private final ExecutorService _workerPool;
-  private final Cache<Long, Long> _cancelledRequests;
+  private final Cache<Long, Long> _cancelledRequests = CacheBuilder.newBuilder()
```
Comment (xiangfu0) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java:55]: Shall it still be tunable?

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java [hunk 4]
```diff
@@ -128,21 +129,21 @@ protected TransferableBlock getNextBlock() {
...
-                OperatorUtils.getMetadataFromOperatorStats(_opChainStats.getOperatorStatsMap()));
-            _exchange.send(eosBlockWithStats);
-          } else {
-            _exchange.send(transferableBlock);
-          }
-          return transferableBlock;
+      if (transferableBlock.isNoOpBlock()) {
+        return transferableBlock;
+      } else if (transferableBlock.isEndOfStreamBlock()) {
+        if (transferableBlock.isSuccessfulEndOfStreamBlock()) {
```
Comment (xiangfu0) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java:135]: seems like only `isEndOfStreamBlock` is special handling, other cases you always `return transferableBlock;`

**Review by xiangfu0 — APPROVED** [review]
overall lgtm

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java (hunk 3 above)
Comment (walterddr) [reply pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java:55]: shouldn't be a problem. cancellation signal are bounded by the control flow not the mailbox data flow, which has a much smaller timeout

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java (hunk 4 above)
Comment (walterddr) [reply pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxSendOperator.java:135]: yes you are correct. code-wise they are identical. here i just separate them to distinguish signal block vs. data block for readability.

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java (hunk 3 above)
Comment (xiangfu0) [reply pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java:55]: Hmm, what's the behavior when you have a very long timeout query? Or we should limit it from both control/data flows

**Pull request merged**